src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,416 @@
+	struct oval_state_content *evr_content = oval_state_content_new(def_model);
+	oval_state_content_set_entity(evr_content, evr_entity);
+
+	char *state_id = get_oval_id_string("ste", stateNo);
+	struct oval_state *state = oval_definition_model_get_new_state(def_model, state_id);
+	free(state_id);
+	oval_state_set_comment(state, attributes->full_package_name);
+	oval_state_set_subtype(state, OVAL_LINUX_RPM_INFO);
+	oval_state_set_operator(state, OVAL_OPERATOR_AND);
+	oval_state_set_version(state, 1);
+	oval_state_add_content(state, state_content);
+	oval_state_add_content(state, evr_content);
+
+	return state;
+}
+
+static struct oval_test *get_new_rpminfo_test_for_cvrf(struct oval_definition_model *def_model,
+		struct cvrf_rpm_attributes *attributes, unsigned int testNo) {
+
+	char *test_id = get_oval_id_string("tst", testNo);
+	struct oval_test *rpm_test = oval_test_new(def_model, test_id);
+	free(test_id);
+	oval_test_set_subtype(rpm_test,OVAL_LINUX_RPM_INFO);
+	oval_test_set_version(rpm_test, 1);
+	oval_test_set_check(rpm_test, OVAL_CHECK_AT_LEAST_ONE);
+	oval_test_set_existence(rpm_test, OVAL_AT_LEAST_ONE_EXISTS);
+
+	oval_test_add_state(rpm_test, get_new_oval_state_for_cvrf(def_model, attributes, testNo));
+	oval_test_set_object(rpm_test, get_new_oval_object_for_cvrf(def_model, attributes, testNo));
+
+	return rpm_test;
+}
+
+static struct oval_definition *get_new_oval_definition_for_cvrf(struct oval_definition_model *def_model,
+		struct cvrf_rpm_attributes *attributes, unsigned int index) {
+
+	char *definition_id = get_oval_id_string("def", index);
+	struct oval_definition *definition = oval_definition_model_get_new_definition(def_model, definition_id);
+	free(definition_id);
+	oval_definition_set_version(definition, 1);
+	oval_definition_set_title(definition, "CVRF RPM Vulnerability Test");
+
+	struct oval_criteria_node *criteria = oval_criteria_node_new(def_model, OVAL_NODETYPE_CRITERIA);
+	oval_definition_set_criteria(definition, criteria);
+
+	struct oval_criteria_node *criterion = oval_criteria_node_new(def_model, OVAL_NODETYPE_CRITERION);
+	oval_criteria_node_set_test(criterion, get_new_rpminfo_test_for_cvrf(def_model, attributes, index));
+	char *comment = oscap_sprintf("Check for vulnerability of package %s", attributes->rpm_name);
+	oval_criteria_node_set_comment(criterion, comment);
+	oval_criteria_node_add_subnode(criteria, criterion);
+
+	return definition;
+}
+
+int cvrf_session_construct_definition_model(struct cvrf_session *session) {
+	struct oval_definition_model *def_model = session->def_model;
+	struct oscap_string_iterator *product_ids = cvrf_session_get_product_ids(session);
+	unsigned int index = 1;
+
+	while (oscap_string_iterator_has_more(product_ids)) {
+		const char *product_id = oscap_string_iterator_next(product_ids);
+		struct cvrf_rpm_attributes *attributes = parse_rpm_attributes_from_cvrf_product_id(session, product_id);
+		get_new_oval_definition_for_cvrf(def_model, attributes, index);
```

Comment:
May I suggest a better name for the same function then?

`create_oval_definition_for_cvrf_rpm_attributes`. That implies that something is created and you either get the ref or you don't, either way it's created.